tests/plugins/test_pulp_tag.py

```diff
@@ -0,0 +1,254 @@
+"""
+Copyright (c) 2017 Red Hat, Inc
+All rights reserved.
+
+This software may be modified and distributed under the terms
+of the BSD license. See the LICENSE file for details.
+"""
+
+from __future__ import unicode_literals
+
+import os
+import sys
+import json
+
+from osbs.build.build_response import BuildResponse
+from atomic_reactor.core import DockerTasker
+from atomic_reactor.inner import DockerBuildWorkflow
+from atomic_reactor.plugin import PostBuildPluginsRunner, PluginFailedException
+from atomic_reactor.util import ImageName
+from atomic_reactor.build import BuildResult
+from atomic_reactor.plugins.build_orchestrate_build import (OrchestrateBuildPlugin,
+                                                            WORKSPACE_KEY_BUILD_INFO)
+from atomic_reactor.constants import PLUGIN_PULP_TAG_KEY
+
+try:
+    if sys.version_info.major > 2:
+        # importing dockpulp in Python 3 causes SyntaxError
+        raise ImportError
+
+    import dockpulp
+except (ImportError):
+    dockpulp = None
+
+import pytest
+from flexmock import flexmock
+from tests.constants import INPUT_IMAGE, SOURCE, MOCK
+if MOCK:
+    from tests.docker_mock import mock_docker
+
+
+class X(object):
+    image_id = INPUT_IMAGE
+    git_dockerfile_path = None
+    git_path = None
+    base_image = ImageName(repo="qwe", tag="asd")
+
+
+class BuildInfo(object):
+    def __init__(self, v1_image_id=None):
+        annotations = {'meta': 'test'}
+        if v1_image_id:
+            annotations['v1-image-id'] = json.dumps(v1_image_id)
+
+        self.build = BuildResponse({'metadata': {'annotations': annotations}})
+
+
+def prepare(check_repo_retval=0, v1_tags={}):
```

Comment:
There's naming confusing between tags and image IDs here. These `v1_tags` get set as the value of the "v1-image-id" annotation.

An image ID looks like: `"002c95e6c2b4a56113d0d5cdeab7d203646125287b5f753345b5e6a4491c2fec"`.

The tags applied to that image ID might look like `["7.1-29", "7.1", "latest"]`.

This would result in an updateRepo call with a parameter like: `{"tags": "7.1-29,7.1,latest:002c95e6c2b4a56113d0d5cdeab7d203646125287b5f753345b5e6a4491c2fec"}`

(Yes, that way round... no, I don't know why)

So this parameter should be named `v1_image_ids`.